manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/persistence/models/Processor.java

```diff
@@ -31,9 +33,26 @@
                 query = "select distinct (p) from Processor_V2 p left join fetch p.bridge left join fetch p.conditions where p.bridge.id=:bridgeId and p.bridge.customerId=:customerId order by p.submittedAt desc"),
         @NamedQuery(name = "PROCESSOR_V2.findByIdBridgeIdAndCustomerId",
                 query = "from Processor_V2 p left join fetch p.bridge left join fetch p.conditions where p.id=:id and p.bridge.id=:bridgeId and p.bridge.customerId=:customerId"),
+        @NamedQuery(name = "PROCESSOR_V2.findByIdsWithBridgeAndConditions",
+                query = "select distinct (p) from Processor_V2 p left join fetch p.bridge left join fetch p.conditions where p.id in (:ids)"),
         @NamedQuery(name = "PROCESSOR_V2.countByBridgeIdAndCustomerId",
                 query = "select count(p.id) from Processor_V2 p where p.bridge.id=:bridgeId and p.bridge.customerId=:customerId")
 })
+@NamedNativeQueries({
+        @NamedNativeQuery(name = "PROCESSOR_V2.findProcessorIdByShardIdToDeployOrDelete",
+                query = "select p.id " +
+                        "from Processor_V2 p " +
+                        "left join Bridge_V2 b on p.bridge_id = b.id " +
+                        "left join (" +
+                        "  select processor_id, count(*) as conditions_count from Condition c where component='MANAGER' group by c.processor_id" +
+                        ") cp_counts on cp_counts.processor_id=p.id " +
+                        "left join (" +
+                        "  select processor_id, count(*)  as conditions_count from Condition c where component='MANAGER' and status='TRUE' group by c.processor_id" +
+                        "    ) cp_ready_counts on cp_ready_counts.processor_id=p.id " +
+                        "where " +
+                        "b.shard_id = :shardId and " +
+                        "cp_counts.conditions_count = cp_ready_counts.conditions_count")
+})
```

Comment:
I think this query could be simplified with something like (just by heart, not tested at all)

```
@NamedNativeQueries({
        @NamedNativeQuery(name = "PROCESSOR_V2.findProcessorIdByShardIdToDeployOrDelete",
                query = "select p.id " +
                        "from Processor_V2 p " +
                        "left join Bridge_V2 b on p.bridge_id = b.id " +
                        "left join (" +
                        "  select processor_id, count(*) from Condition c where component='MANAGER' and status != 'TRUE' group by c.processor_id having count(*)=0" +
                        "    ) cp_ready on cp_ready.processor_id=p.id " +
                        "where " +
                        "b.shard_id = :shardId")
})
```

Reply:
Thanks for the suggestion. Whilst it did not work when all `MANAGER` `Condition`'s are `TRUE` it pointed my towards the following which has one less `join` and one less sub-query so is an improvement on my original:
```
@NamedNativeQueries({
        @NamedNativeQuery(name = "PROCESSOR_V2.findProcessorIdByShardIdToDeployOrDelete",
                query = "select p.id " +
                        "from Processor_V2 p " +
                        "left join Bridge_V2 b on p.bridge_id = b.id " +
                        "left join (" +
                        "  select processor_id, count(*) as incomplete_count from Condition c where component='MANAGER' and status != 'TRUE' group by c.processor_id " +
                        "    ) cp on cp.processor_id = p.id " +
                        "where " +
                        "(cp.incomplete_count = 0 or cp.incomplete_count is null) and " +
                        "b.shard_id = :shardId")
```